test/autogen/self_improve_agent.md

```diff
@@ -0,0 +1,53 @@
+# Self-Improving API Agent
+This code describes a scenario where an API agent is designed to self-improve
+and interact with a user. The API agent tries to answer a given question using the provided API or suggests a new API if the existing one is not suitable.
+
+The code is divided into two modes: User mode and Dev mode. The user mode provides a basic interaction between the API agent and the user agent, while the Dev mode involves an expert agent for verifying the correctness of the answers provided by the API agent.
+
+## User mode
+In User mode, the API agent use `receive` the function call to receive questions from users and answer the questions. It attempts to answer the question with existing apis and will try to add new apis if the question cannot be answered by existing ones properly.
+
+```python
+# An API agent that performs self-improvement by default
+api_agent = APIAgent("api_agent", self_improve=True, api="api.json")
+
+# User mode
+user = HumanAgent("human user")
+# Question from the user
+question = "What if we must go from node 1 to node 2?"
+# the api_agent will send the answer to user
+api_agent.receive(question, user)
+```
+
```

Comment:
Describe what happens after the `receive` call.

Reply:
To be more specific, do we want to assume what experience happens after `receive()`? For example, like ChatGPT, like AutoGPT, or neither, or both.

Reply:
Updated accordingly.